analyzer/src/main/kotlin/managers/GoMod.kt

```diff
@@ -327,6 +327,8 @@ private val PSEUDO_VERSION_REGEXES = listOf(
     "^v0.0.0-[\\d]{14}-(?<sha1>[0-9a-f]+)$".toRegex(),
     // base version is a release version, e.g. v1.2.4-0.20191109021931-daa7c04131f5.
     "^v[0-9]+\\.[0-9]+\\.[0-9]+-0\\.[\\d]{14}-(?<sha1>[0-9a-f]+)$".toRegex(),
+    // base version is a pre-release version, e.g. v1.2.4-pre.0.20191109021931-daa7c04131f5.
+    "^v[0-9]+\\.[0-9]+\\.[0-9]+-pre.0\\.[\\d]{14}-(?<sha1>[0-9a-f]+)$".toRegex(),
```

Comment:
Can't we simply make the `pre.` part optional in the expression above?

Reply:
Yes we can. I however prefer it as-is, because to me it is easier to understand and also nice to have one separate entry per format in the spec. Ok with keeping it as-is?

Reply:
How about then at least extracting some repeating parts like `[\\d]{14}-(?<sha1>[0-9a-f]+)$` to a common variable?

Reply:
Done in a separate commit on top.